test/transaction.cpp

```diff
@@ -0,0 +1,154 @@
+           Transaction(toInt(tObj["value"]), toInt(tObj["gasPrice"]), toInt(tObj["gasLimit"]), Address(tObj["to"].get_str()), importData(tObj), toInt(tObj["nonce"]), Secret(tObj["secretKey"].get_str()));
+
+        //cnote << "Readed RLP: " << toHex(rlpReaded);
+        //cnote << "Sender: " << tr.sender();
+        //cnote << "Transaction RLP: " << toHex(tr.rlp());
+
+        Transaction trFromRLP = Transaction(rlpReaded);
+
+        //Check the fields restored from RLP to original fields
+        if(tr.data() != trFromRLP.data())
+            BOOST_ERROR("Data in given RLP not matching the Transaction data!");
+        if(tr.value() != trFromRLP.value())
+            BOOST_ERROR("Value in given RLP not matching the Transaction value!");
+        if(tr.gasPrice() != trFromRLP.gasPrice())
+            BOOST_ERROR("GasPrice in given RLP not matching the Transaction gasPrice!");
+        if(tr.gas() != trFromRLP.gas())
+            BOOST_ERROR("Gas in given RLP not matching the Transaction gas!");
+        if(tr.nonce() != trFromRLP.nonce())
+            BOOST_ERROR("Nonce in given RLP not matching the Transaction nonce!");      
+        if(tr.receiveAddress() != trFromRLP.receiveAddress())
+            BOOST_ERROR("Receive address in given RLP not matching the Transaction 'to' address!");
+        if(tr.sender() != trFromRLP.sender())
+            BOOST_ERROR("Transaction sender address in given RLP not matching the Transaction 'vrs' signature!");
+        if(tr.sender() !=  addressReaded || trFromRLP.sender() != addressReaded)
+            BOOST_ERROR("Signature address of sender does not match given sender address!");
+    }
+}
+} }// Namespace Close
+
+
+BOOST_AUTO_TEST_SUITE(TransactionTests)
+
+/*BOOST_AUTO_TEST_CASE(ttExample)
+{
+    dev::test::executeTests("ttExample", "/TransactionTests", dev::test::doTransactionTests);
+}*/
+
+BOOST_AUTO_TEST_CASE(ttCreateTest)
+{
+    for (int i = 1; i < boost::unit_test::framework::master_test_suite().argc; ++i)
+    {
+        string arg = boost::unit_test::framework::master_test_suite().argv[i];
+        if (arg == "--createtest")
+        {
+            if (boost::unit_test::framework::master_test_suite().argc <= i + 2)
+            {
+                cnote << "usage: ./testeth --createtest <PathToConstructor> <PathToDestiny>\n";
+                return;
+            }
+            try
+            {
+                cnote << "Populating tests...";
+                json_spirit::mValue v;
+                string s = asString(dev::contents(boost::unit_test::framework::master_test_suite().argv[i + 1]));
+                BOOST_REQUIRE_MESSAGE(s.length() > 0, "Content of " + (string)boost::unit_test::framework::master_test_suite().argv[i + 1] + " is empty.");
+                json_spirit::read_string(s, v);
+                dev::test::doTransactionTests(v, true);
```

Comment:
this is not working right now, ince you have no filler files, and you need to write some code to produce test files out of json files using the _fillin bool in doTransactionTests